src/main/java/com/aws/greengrass/clientdevices/auth/metrics/ClientDeviceAuthMetrics.java

```diff
@@ -18,6 +18,10 @@
 public class ClientDeviceAuthMetrics {
 
     private final AtomicLong certSubscribeSuccess = new AtomicLong();
+    private final AtomicLong certSubscribeError = new AtomicLong();
```

Comment:
Take a look at the event emitter infrastructure that we have. We need to be using that in order to keep this code de-coupled.

Application code should emit a metric event containing the event. This code can listen for those events and handle them appropriately.

I also want to re-visit metric names here to make sure we're getting the ones we want. But, the good news is, if we code it up properly, we can make changes easily without needing to make a lot of changes.